File: codingsample/202008/20200822/Island_of_Knowledge_5_boxBlur.py
def boxBlur(image):
    output = [[0]*(len(image[0])-2) for i in range(len(image)-2)]
    for i in range(len(output)):
        for j in range(len(output[0])):
            output[i][j] = int(sum(image[i][q1]+image[i+1][q1]+image[i+2][q1] for q1 in range(j,j+3))/9)
    return output

File: codingsample/202008/20200822/Island_of_Knowledge_5_boxBlur.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h> 

int sum3(int a,int b, int c){
    return a+b+c;
}

arr_arr_integer boxBlur(arr_arr_integer image) {
    arr_arr_integer output = alloc_arr_arr_integer(image.size-2);
    int i,j,k,temp;
    for(i=0;i<output.size;i++){
        output.arr[i] = alloc_arr_integer(image.arr[0].size-2);
        for(j=0;j<output.arr[i].size;j++){
            temp=0;
            for(k=j;k<=j+2;k++) temp+=sum3(image.arr[i].arr[k],image.arr[i+1].arr[k],image.arr[i+2].arr[k]);
            output.arr[i].arr[j] = temp/9;
        }
    }
    return output;
}


